app/controllers/avo/actions_controller.rb

```diff
@@ -48,6 +48,7 @@ def set_action
         view: :new, # force the action view to in order to render new-related fields (hidden field)
         arguments: BaseAction.decode_arguments(params[:arguments] || params.dig(:fields, :arguments)) || {}
       )
+      @fields = @action.get_fields
```

Comment:
This run every time that `set_action` is called. `set_action`  is called on `:show` and `:handle` (`before_action :set_action, only: [:show, :handle]`).

Since we only want to use `@fields` on `show` lets move this line to `show` method to avoid the unnecessary call on `handle`

Reply:
Got it.